runtime/ftplugin/help_toc.vim

```diff
@@ -0,0 +1,66 @@
+if !exists('g:no_plugin_maps')
+  function! s:show_toc() abort
+    if !exists('b:qf_toc')
```

Comment:
Can the qf title be used? 

> I assume getting the first location list item's bufnr, but I don't like that since quickfix is an unreliable source for state information.

That sounds good too. Unreliable because the user could edit it? The worst that happens is the TOC is recalculated, so the stakes are low...